### PR TITLE
feat: add provider version to state

### PR DESCRIPTION
**Pull request by argoyle**
This seems to work at least. Not sure it's the best or preferred way to pass the `providerVersion` around or set it in the state. Not sure we need it in the updater since the cluster and instance groups will be updated anyway which would trigger the updater as well, right. Feel free to change it or add comments on how you would like it to be done.

### CODE REVIEW

**Comment by argoyle** [comment]
Running this version based on the main-branch gives me the following error though:
```
╷
│ Error: missing expected [
│
│   with module.k8s.kops_cluster.k8s,
│   on .terraform/modules/k8s/k8s.tf line 20, in resource "kops_cluster" "k8s":
│   20: resource "kops_cluster" "k8s" {
│
╵
```

Basing it on the 1.23.4-tag works.

**Comment by dkulchinsky** [comment]
hitting the same with `1.23.5-alpha.1`, but is fine with `1.23.4`

```
│ Error: missing expected [
│
│   with module.kubernetes_kops_aws_cluster_awsuse2["sbx-awsuse2"].kops_cluster.cluster,
│   on ../modules/kubernetes-cluster/kops-aws/kops.tf line 15, in resource "kops_cluster" "cluster":
│   15: resource "kops_cluster" "cluster" {
```

/cc @eddycharly

**Comment by argoyle** [comment]
For me it was an existing cluster. Will try using a new cluster as well.

**Comment by eddycharly** [comment]
Ok, i will add migration code.

**Comment by argoyle** [comment]
Creating a new cluster seems to work fine

**Comment by dkulchinsky** [comment]
@eddycharly same for me as reported in https://github.com/eddycharly/terraform-provider-kops/issues/597, also on an existing cluster

**Comment by eddycharly** [comment]
@argoyle @dkulchinsky this should fix the issue https://github.com/eddycharly/terraform-provider-kops/pull/603

**Comment by eddycharly** [comment]
@argoyle Thanks for the PR, I'll look at it on monday.

**Comment by eddycharly** [comment]
@argoyle `v1.23.5-alpha.2` is being released, with the fix from #603

**Comment by argoyle** [comment]
Just rebased from main and rebuilt and it seems to work as expected. Thanks!

**Comment by eddycharly** [comment]
@argoyle do we need to set the provider version in the cluster and instance group resources ?

**Comment by eddycharly** [comment]
@argoyle see https://github.com/eddycharly/terraform-provider-kops/pull/606 for the version info.
It should make the implementation easier.

**Comment by argoyle** [comment]
Not sure if we need it in cluster and instance group. It might be enough to do the updater.

**Comment by argoyle** [comment]
I can try to do the change after #606 is merged

**Comment by eddycharly** [comment]
Merged.
Maybe the updater is enough indeed.

File: pkg/resources/ClusterUpdater.go [hunk 1]
```diff
@@ -10,13 +10,13 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func ClusterUpdater() *schema.Resource {
+func ClusterUpdater(providerVersion string) *schema.Resource {
 	return &schema.Resource{
 		CreateWithoutTimeout: ClusterUpdaterCreateOrUpdate,
 		ReadContext:          schema.NoopContext,
```
Comment (eddycharly) [inline pkg/resources/ClusterUpdater.go:16]: Why not implement `Read`, set the version there, and let terraform do the rest ?

Comment (argoyle) [reply pkg/resources/ClusterUpdater.go:16]: I tried that once and it seemed to not update on changes then. I might have done something wrong though. 😄

Comment (eddycharly) [reply pkg/resources/ClusterUpdater.go:16]: @argoyle https://github.com/eddycharly/terraform-provider-kops/pull/607 WDYT ?

Comment (argoyle) [reply pkg/resources/ClusterUpdater.go:16]: This matches the outcome when I tried the Read-implementation myself as well. 😊

Comment (eddycharly) [reply pkg/resources/ClusterUpdater.go:16]: Right, wrong logic, we can't refresh the state on read...
Pushed another commit, hope this is better :crossed_fingers:

Comment (argoyle) [reply pkg/resources/ClusterUpdater.go:16]: Yeay! When I finally managed to understand how the ldflags should be set when the variable is not in main it seems to work correctly.

You need to update the ldflags in goreleaser-config again for it to work though:
```
    ldflags:
      - -s -w
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildVersion={{.Version}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildCommit={{.Commit}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildDate={{.Date}}'
      - -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuiltBy=goreleaser'
```

I also change the build make-target to be able to do `make PROVIDER_VERSION=1.23.5`
```
.PHONY: build
build: gen
	@CGO_ENABLED=0 go build -ldflags="-s -w -X 'github.com/eddycharly/terraform-provider-kops/pkg/version.BuildVersion=v${PROVIDER_VERSION}'" ./cmd/terraform-provider-kops
```

Comment (eddycharly) [reply pkg/resources/ClusterUpdater.go:16]: Thanks for cracking it down ! :+1: 
I updated the PR and will release one more alpha.

**Pull request closed without merging**